nixos/modules/services/web-servers/garage.nix

```
@@ -71,6 +64,19 @@ in
   };
 
   config = mkIf cfg.enable {
+
+    assertions = [
+      # We removed our module-level default for replication_mode. If a user upgraded
+      # to garage 1.0.0 while relying on the module-level default, they would be left
+      # with a config which evaluates and builds, but then garage refuses to start
+      # because either replication_factor or replication_mode is required.
+      # This assertion can be removed in NixOS 24.11, when all users have been warned once.
+      {
+        assertion = (cfg.settings ? replication_factor || cfg.settings ? replication_mode) || lib.versionOlder cfg.package "1.0.0";
+        message = "Garage 1.0.0 requires an explicit replication factor to be set.\nPlease set replication_factor to 1 explicitly to preserve the previous behavior.\nhttps://git.deuxfleurs.fr/Deuxfleurs/garage/src/tag/v1.0.0/doc/book/reference-manual/configuration.md#replication_factor";
```

Use multiline strings
```suggestion
        message = ''
          Garage requires an explicit replication factor to be set since version 1.0.0.
        
          Please set `services.garage.settings.replication_factor` to 1 explicitly to preserve the previous behavior.
          See https://git.deuxfleurs.fr/Deuxfleurs/garage/src/tag/v1.0.0/doc/book/reference-manual/configuration.md#replication_factor for more info.
        '';
```